blockchain/blockhashHandler_test.go

```diff
@@ -0,0 +1,191 @@
+		args := createMockArgsBlockhashHandler()
+		handler, err := NewBlockhashHandler(args)
+
+		assert.False(t, check.IfNil(handler))
+		assert.Nil(t, err)
+	})
+	t.Run("process loop should start", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		handler, _ := NewBlockhashHandler(args)
+		time.Sleep(2 * time.Second) // wait for the process loop to start
+
+		assert.True(t, handler.loopStatus.IsSet())
+	})
+}
+
+func TestGetBlockTimestampByHash_Success(t *testing.T) {
+	expectedTimestamp := int(time.Now().Unix())
+
+	args := createMockArgsBlockhashHandler()
+	args.httpClientWrapper = &testsCommon.HTTPClientWrapperStub{
+		GetHTTPCalled: func(ctx context.Context, url string) ([]byte, int, error) {
+			block := data.Block{Timestamp: expectedTimestamp}
+			blockBytes, _ := json.Marshal(&block)
+			return blockBytes, http.StatusOK, nil
+		},
+	}
+	bh, _ := NewBlockhashHandler(args)
+	actualTimestamp, err := bh.GetBlockTimestampByHash(context.Background(), "blockHash")
+
+	assert.Nil(t, err)
+	assert.NotNil(t, actualTimestamp)
+	assert.Equal(t, expectedTimestamp, actualTimestamp)
+}
+
+func TestGetBlockTimestampByHash_HTTPError(t *testing.T) {
+	args := createMockArgsBlockhashHandler()
+	args.httpClientWrapper = &testsCommon.HTTPClientWrapperStub{
+		GetHTTPCalled: func(ctx context.Context, url string) ([]byte, int, error) {
+			return nil, http.StatusBadRequest, fmt.Errorf("Bad Request")
+		},
+	}
+	bh, _ := NewBlockhashHandler(args)
+
+	actualTimestamp, err := bh.GetBlockTimestampByHash(context.Background(), "blockHash")
+
+	assert.NotNil(t, err)
+	assert.Equal(t, actualTimestamp, 0)
+}
+
+func TestGetBlockTimestampByHash_InvalidJSON(t *testing.T) {
+	args := createMockArgsBlockhashHandler()
+	args.httpClientWrapper = &testsCommon.HTTPClientWrapperStub{
+		GetHTTPCalled: func(ctx context.Context, url string) ([]byte, int, error) {
+			return []byte(`{"timestamp": "invalid"}`), http.StatusOK, nil
+		},
+	}
+
+	bh, _ := NewBlockhashHandler(args)
+
+	actualTimestamp, err := bh.GetBlockTimestampByHash(context.Background(), "blockHash")
+
+	assert.NotNil(t, err)
+	assert.Equal(t, actualTimestamp, 0)
+}
+
+func TestGetBlockTimestampByHash_ExpiredBlock(t *testing.T) {
+	args := createMockArgsBlockhashHandler()
+	args.httpClientWrapper = &testsCommon.HTTPClientWrapperStub{
+		GetHTTPCalled: func(ctx context.Context, url string) ([]byte, int, error) {
+			block := data.Block{Timestamp: int(time.Now().Add(-2 * time.Second).Unix())}
+			blockBytes, _ := json.Marshal(&block)
+			return blockBytes, http.StatusOK, nil
+		},
+	}
+
+	bh, _ := NewBlockhashHandler(args)
+	currentTime := time.Now()
+	bh.getTimeHandler = func() time.Time {
+		return currentTime.Add(time.Second * 1120)
+	}
+
+	timestamp, err := bh.GetBlockTimestampByHash(context.Background(), "blockHash")
+
+	assert.Nil(t, err)
+	assert.NotNil(t, timestamp)
+	// check bh.blockhashes does not containt the blockHash
+	_, ok := bh.blockhashes["blockHash"]
```

Comment:
not concurrent safe.. same note as above, export a method to return a copy of the map, while protected by a mutex